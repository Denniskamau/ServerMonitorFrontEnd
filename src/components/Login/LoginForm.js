import React from 'react';
import { Field } from 'redux-form';
import formcomponent from '../formcomponent'


export const LoginForm = ({ handleSubmit, onSubmit, props }) => {

  return (
    <div>
      <h1>login</h1>
    
      <form
        className="loginForm"
        onSubmit={handleSubmit(onSubmit)}
      >
      <Field name="email" label="Email" component={formcomponent} type="email" placeholder="Email" />
      <Field name="password" label="Password" component={formcomponent} type="password" placeholder="Password" />

        <button type="submit" className="loginBtn">
          Submit
        </button>
      </form>
      
    </div>
  );
}

export default LoginForm