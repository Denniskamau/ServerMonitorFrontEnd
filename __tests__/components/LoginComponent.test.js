import React from 'react'
import { shallow, mount, render } from 'enzyme'
import LoginForm from '../../src/components/Login/LoginContainer'

describe('render login', ()=>{
    let wrapper
    beforeEach(()=>{
        wrapper = shallow(<LoginForm/>)
        // console.log('wrapper',JSON.stringify(wrapper))
    })
    it('should render correctly', ()=>{
  
        expect(wrapper).toMatchSnapshot();
    })
    // it('should find login form', () => {
    //     expect(wrapper.find('form').length).toEqual(1);
    // });

    // it('email must be provided', ()=>{
    //     const values = {
    //         email:'',
    //         password:'password'
    //     }
    //     const result = wrapper.instance().submit(values.email,values.password)
    //     console.log('result is',JSON.stringify(result))
    //     expect(result).toEqual('Required')
    // })
    // it('password must be provided', ()=>{
    //     const values = {
    //         email:'test@test.com',
    //         password:''
    //     }
    //     const result = wrapper.instance().submit(values.email,values.password)
    //     console.log('result is', result.errors)
    //     expect(result).toEqual('Required')
    // })
    // it('error to be undefined if all values are provided', ()=>{
    //     const values = {
    //         email:'test@test.com',
    //         password:'password'
    //     }
    //     const result = wrapper.instance().submit()
    //     console.log('result is', result)
    //     expect(result).toEqual('Hello')
    // })
})

// describe('handle submit', ()=>{
//     it('should call submit function', () => {
//         const component = shallow(<LoginForm/>)
//         const submit = jest.fn()
//         submit()
//         component.find('[form]').simulate('click')
//         expect(submit).toHaveBeenCalled()
//     });
//     it('should get instance of submit', ()=>{
//         const component = shallow(<LoginForm/>)
//         component.submit = jest.fn()
//         const submit = jest.fn((val)=> {
//             return val
//         })
//         const result = submit('Hello') 
//         expect(component.submit).toEqual('Hello')
       
//     })
// })
