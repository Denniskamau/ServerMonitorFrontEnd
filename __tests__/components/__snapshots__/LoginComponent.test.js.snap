// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "LoginForm",
      "getFormState": [Function],
      "initialValues": undefined,
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "LoginForm",
        "getFormState": [Function],
        "initialValues": undefined,
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`render login should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "LoginForm",
      "getFormState": [Function],
      "initialValues": undefined,
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "LoginForm",
        "getFormState": [Function],
        "initialValues": undefined,
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
